scipy/stats/stats.py

```diff
@@ -3692,6 +3692,103 @@ def f_oneway(*args, axis=0):
+                Approximation for ANOVA under Variance Heterogeneity.” Journal
+                of Educational Statistics, vol. 19, no. 2, 1994, pp. 91–101.
+                JSTOR, www.jstor.org/stable/1165140. Accessed 12 Sept. 2020.
+    Examples
+        --------
+        >>> from scipy.stats import AlexanderGovern, Kruskal, F_oneway
+
+        >>> young = [482.43, 484.36, 488.84, 495.15, 495.24, 502.69, 504.62,
+        ...         518.29, 519.1, 524.1, 524.12, 531.18, 548.42, 572.1,
+        ...         584.68, 609.09, 609.53, 666.63, 676.4]
+        >>> middle = [335.59, 338.43, 353.54, 404.27, 437.5, 469.01, 485.85,
+        ...          487.3, 493.08, 494.31, 499.1, 800]
+        >>> old = [519.01, 528.5, 530.23, 536.03, 538.56, 538.83, 557.24,
+        ...        558.61, 558.95, 565.43, 586.39, 594.69, 629.22, 645.69,
+        ...        691.84]
+
+        >>> A, p = AlexanderGovern(young, middle, old)
+        >>> A, p
+        (6.941146076872535, 0.03109920451449096)
+    """
+    if len(args) < 2:
+        raise TypeError(f"2 or more inputs required, got {len(args)}")
+
+    args = [np.asarray(arg, dtype=float) for arg in args]
+
+    # The following formula numbers reference the equation described on
+    # page 92 by Alexander, Govern. Formulas 5, 6, and 7 describe other
+    # tests that serve as the basis for equation (8) but are not needed
+    # to perform the test.
+
+    # (1) determine standard errors for each sample
+    standard_errors = [np.std(arg, ddof=1) / np.sqrt(len(arg)) for arg in args]
+
+    # precalculate weighted sum for following step
+    weight_denom = np.sum(1 / np.square(standard_errors))
+
+    # (2) define a weight for each samlple
+    weights = [(1 / s**2) / weight_denom for s in standard_errors]
+
+    # precalculate means of each sample
+    means = np.asarray([np.mean(arg) for arg in args])
+
+    # (3) determine variance-weighted estimate of the common mean
+    var_w = np.sum(weights * means)
+
+    # (4) determine one-sample t statistic for each group
+    t_stats = [((mean - var_w)/s) for mean, s in zip(means, standard_errors)]
+    # calculate parameters to be used in transformation
+    v = [len(k) - 1 for k in args]
+    a = [v_i - .5 for v_i in v]
+    b = [48 * a_i**2 for a_i in a]
+    c = [((a_i * np.log(1 + (t_i ** 2)/v_i))**.5)
+         for a_i, t_i, v_i in zip(a, t_stats, v)]
+
+    # (8) perform a normalizing transformation on t statistic
+    z = [(c_i + ((c_i**3 + 3*c_i)/b_i) -
+          ((4*c_i**7 + 33*c_i**5 + 240*c_i**3 + 855*c_i) /
+           (10*b_i**2 + 8*b_i*c_i**4 + 1000*b_i)))
+         for c_i, b_i in zip(c, b)]
+
+    # (9) calculate statistic
+    A = np.sum(np.square(z))
+
+    # "[the p value is determined from] central chi-square random deviates
+    # with n_i - 1 degrees of freedom". Alexander, Govern (94)
+    p = distributions.chi2.sf(A, len(args) - 1)
+    return AlexanderGovernResult(A, p)
+
+
```

Comment:
This an alternative version of the test I based on Sam's awesome approach that follows a slightly different strategy by running the test on each sample individually (after communal values like standard error have been calculated) rather than step-by-step. The docstring is Sam's with an added source that I used.

I tried to take advantage of vectorization which cuts computation time by half on my computer but is significantly slower on Sam's as proven by `line_profiler` and numerous `timeit` tests. I imagine this could be due to the different processors, OSes, or (as far as I know) the chaos effect - but I'm no Python optimization guru. If anybody has suggestions on why this might be or has any general feedback on this test, I'd love to hear it!

Thanks!

```suggestion
def alexandergovern_alt(*args):
    """
    Performs the Alexander Govern test.
    The Alexander-Govern approximation tests the equality of k independent
    means in the face of heterogeneity of variance. The test is applied to
    samples from two or more groups, possibly with differing sizes.
    Parameters
    ----------
    sample1, sample2, ... : array_like
        The sample measurements for each group.  There must be at least
        two arguments.
    Returns
    -------
    statistic : float
        The computed A statistic of the test.
    pvalue : float
        The associated p-value from the chi-squared distribution.
    References
        ----------
        ... [1] Alexander, Ralph A., and Diane M. Govern. “A New and Simpler
                Approximation for ANOVA under Variance Heterogeneity.” Journal
                of Educational Statistics, vol. 19, no. 2, 1994, pp. 91–101.
                JSTOR, www.jstor.org/stable/1165140. Accessed 12 Sept. 2020.
        ... [2] Kingsley Ochuko, T., Abdullah, S., Binti Zain, Z., & Soaad Syed
                Yahaya, S. (2015). The Modification and Evaluation of the
                Alexander-Govern Test in Terms of Power. Modern Applied
                Science, 9(13), 1. https://doi.org/10.5539/mas.v9n13p1
    Examples
        --------
        >>> from scipy.stats import alexandergovern_alt, Kruskal, F_oneway
        >>> young = [482.43, 484.36, 488.84, 495.15, 495.24, 502.69, 504.62,
        ...         518.29, 519.1, 524.1, 524.12, 531.18, 548.42, 572.1,
        ...         584.68, 609.09, 609.53, 666.63, 676.4]
        >>> middle = [335.59, 338.43, 353.54, 404.27, 437.5, 469.01, 485.85,
        ...          487.3, 493.08, 494.31, 499.1, 800]
        >>> old = [519.01, 528.5, 530.23, 536.03, 538.56, 538.83, 557.24,
        ...        558.61, 558.95, 565.43, 586.39, 594.69, 629.22, 645.69,
        ...        691.84]
        >>> A, p = alexandergovern_alt(young, middle, old)
        >>> A, p
        (6.941146076872535, 0.03109920451449096)
    """

    def calc_sem(data, mean, length):
        """Calculates standard error using precalculated values"""
        sum_ = np.sum((data - mean) ** 2)
        d = length * (length - 1)
        return (sum_ / d) ** .5

    def calc_z2(var_w, sem, mean, length):
        """Calculates the AG Z^2 statistic for the passed array"""
        # Calc degrees of freedom and t, as described in formula [2](7)
        df = length - 1
        t = (mean - var_w) / sem

        # Calculate a,b,c using evaluating AG's formula [2](9), [2](10)
        a = df - .5
        b = 48 * a ** 2
        c = (a * np.log(1 + t ** 2 / df)) ** .5

        # Calculate the three terms in evaluating AG's Z formula [2](8)
        t0 = c
        t1 = (c ** 3 + 3 * c) / b
        t3 = (4 * c ** 7 + 33 * c ** 5 + 240 * c ** 3 + 855 * c)
        t3d = (10 * b ** 2 + 8 * b * c ** 4 + 1000 * b)

        return (t0 + t1 - t3/t3d) ** 2

    if len(args) < 2:
        raise TypeError(f"2 or more inputs required, got {len(args)}")

    # Convert passed list to ndarray. `fromiter` is faster than `asarray`.
    a = [np.fromiter(A, float) for A in args]

    # Calculate sample constants to avoid recalculating
    lens = np.asarray([len(A) for A in a])
    means = np.asarray([np.mean(A) for A in a])
    standard_error = np.vectorize(calc_sem)(a, means, lens)
    sample_weights = 1 / standard_error ** 2 / np.sum(1 / standard_error ** 2)
    var_w = np.sum(sample_weights * means)

    # Calculate the AG statistic by running calc_z2. Vectorization is faster
    # than list comprehension for larger sample sizes. From this point forth
    # all interactions between samples (standard error, etc) have been cached
    # in the above variables
    z = np.vectorize(calc_z2)(var_w, standard_error, means, lens)
    z = np.sum(z)

    # Calculate P value
    p = stats.distributions.chi2.sf(z, len(a) - 1)

    return AlexanderGovernResult(z, p)
```